kcidb/cache/__init__.py

```diff
@@ -0,0 +1,149 @@
+import hashlib
+import requests
+from email.header import Header
+import urllib.parse
+from google.cloud import storage
+
+class Client:
+    def __init__(self, bucket_name, max_store_size):
+        """
+        Initialize a cache client.
+        Args:
+            bucket_name:    The name of the GCS bucket containing the cache.
+            max_store_size: Maximum size the file can have to be stored.
+        """
+        self.bucket_name = bucket_name
+        self.max_store_size = max_store_size
+        self.client = storage.Client()
+
+    def _extract_content_disposition(self, response):
+        content_disposition = response.headers.get("Content-Disposition")
+
+        if content_disposition is None:
+            parsed_url = urllib.parse.urlparse(response.url)
+            original_filename = parsed_url.path.rsplit('/', 1)[-1]
```

Comment:
What happens if there are no `/` characters? Is that possible? What if there's only one, do we get a usable filename?

Reply:
Okkay